compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -139,11 +182,11 @@ class ScenarioService(
       respObs: StreamObserver[RunScenarioResponse],
   ): Unit = {
     if (enableScenarios) {
-      run(
+      runLive(
         req,
-        respObs,
+        StreamWithoutStatus(respObs),
         { case (ctx, pkgId, name) =>
-          Future.successful(ctx.interpretScenario(pkgId, name))
+          Future(ctx.interpretScenario(pkgId, name))
```

Comment:
If `Context#interpretScenario` is not suppose the `throws` I would keep `Future.successful`. Again use the less powerful feature.